src/basicfuns.jl

```diff
@@ -309,9 +309,28 @@ Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and hand
 non-finite values.
 """
 function logaddexp(x::Real, y::Real)
-    # ensure Δ = 0 if x = y = ± Inf
-    Δ = x == y ? zero(x - y) : abs(x - y)
-    max(x, y) + log1pexp(-Δ)
+    T = float(Base.promote_typeof(x, y))
+    _logaddexp(T(x), T(y))
```

Comment:
you can just use something like
```julia
logaddexp(x, y) = logaddexp(promote(x, y)...)

function logaddexp(x::T, y::T) where {T <: AbstractFloat} 
   # float implementation follows here, can use T
end
```

Reply:
Hmm, I think I need to use `logaddexp(float.(promote(x, y)...)` otherwise there will be some infinite recursion when using non-float values... But I get your point, I'll make the update.